src/openassetio-core/tests/CMakeLists.txt

```diff
@@ -20,9 +20,16 @@ target_sources(openassetio-core-cpp-test-exe
     BatchElementErrorTest.cpp
     ContextTest.cpp
     TraitsDataTest.cpp
+    hostApi/HostInterfaceTest.cpp
+    hostApi/ManagerFactoryTest.cpp
+    hostApi/ManagerImplementationFactoryInterfaceTest.cpp
     hostApi/ManagerTest.cpp
+    log/ConsoleLoggerTest.cpp
+    log/LoggerInterfaceTest.cpp
+    log/SeverityFilterTest.cpp
     managerApi/HostTest.cpp
     managerApi/HostSessionTest.cpp
+    managerApi/ManagerInterfaceTest.cpp
```

Comment:
[Question] This dosen't do any harm, but as I mentioned before, I question if there's much value in these static checks, I'm not sure what we're testing, other than that we haven't forgotten to declare the macro (which has the problem that you're equally likely to forget the test.)

Reply:
In my TDD loop I added the test to a couple of the classes (red), then did the naive solution of manually adding the aliases without using a macro (green), then wrote and used the macro (refactor).  

So by adding the tests I guarded against a bad refactor.  Though that particular benefit only applied to the first couple of classes. Perhaps an explicit test of the macro would be better.

Another benefit is guarding against accidental breakage (e.g. deletion) of the aliases at some point in the future. Though if the aliases were tested implicitly by being used in other bits of code, then that would be sufficient (IMO), but as of this PR they're not used anywhere.

We could perhaps refactor the individual tests into a single templated test?  That might also tick the box of "tests as documentation", since we would maintain a list of classes that have a `Ptr` alias member.

Reply:
I think a single template test would be the best form of this, we could test not just the aliases, but also the existence of the pointer types in general.

Reply:
I think the Python converter stuff will have to be separate, since it may not be compiled if Python support is disabled.

Reply:
Refactored tests in d7632df7

Reply:
Thanks for that, seems much nicer to me.